demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -683,33 +1568,79 @@ int main( int argc,
-                                     S3_PRESIGNED_GET_URL_LENGTH,
-                                     &pPath,
-                                     &pathLen );
+        credentialStatus = getTemporaryCredentials( &transportInterface, sizeof( pDateISO8601 ), &credentialResponse, pDateISO8601, &sigvCreds );
 
-            returnStatus = ( httpStatus == HTTPSuccess ) ? EXIT_SUCCESS : EXIT_FAILURE;
-        }
+        returnStatus = ( credentialStatus == true ) ? EXIT_SUCCESS : EXIT_FAILURE;
 
-        if( returnStatus == EXIT_SUCCESS )
+        if( returnStatus == EXIT_FAILURE )
         {
-            ret = downloadS3ObjectFile( &transportInterface,
-                                        pPath );
-            returnStatus = ( ret == true ) ? EXIT_SUCCESS : EXIT_FAILURE;
+            LogError( ( "Failed to get temporary credentials from AWS IOT CREDENTIALS PROVIDER %s.",
+                        serverHost ) );
         }
 
-        /************************** Disconnect. *****************************/
-
         /* End the TLS session, then close the TCP connection. */
         ( void ) Openssl_Disconnect( &networkContext );
 
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            /* Attempt to connect to the AWS S3 Http server. If connection fails, retry
+             * after a timeout. The timeout value will be exponentially
+             * increased until either the maximum number of attempts or the
+             * maximum timeout value is reached. The function returns
+             * EXIT_FAILURE if the TCP connection cannot be established to the
+             * broker after the configured number of attempts. */
+            returnStatus = connectToServerWithBackoffRetries( connectToS3Server,
+                                                              &networkContext );
+
+            if( returnStatus == EXIT_FAILURE )
+            {
+                /* Log an error to indicate connection failure after all
+                 * reconnect attempts are over. */
+                LogError( ( "Failed to connect to AWS S3 HTTP server %s.",
+                            serverHost ) );
+            }
+
+            /* Define the transport interface. */
+            if( returnStatus == EXIT_SUCCESS )
+            {
+                ( void ) memset( &transportInterface, 0, sizeof( transportInterface ) );
+                transportInterface.recv = Openssl_Recv;
+                transportInterface.send = Openssl_Send;
+                transportInterface.pNetworkContext = &networkContext;
+            }
+
+            /******************** Download S3 Object File. **********************/
+
+            if( returnStatus == EXIT_SUCCESS )
+            {
+                /* Retrieve the path location from AWS_S3_OBJECT_URL. This
+                 * function returns the length of the path without the query into
+                 * pathLen, which is left unused in this demo. */
+                httpStatus = getUrlPath( AWS_S3_OBJECT_URL,
```

Comment:
Same question here about whether extraction of URL path is required when it is known at compile time